Sources/MusicPlayer/Utilities/Glib.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  Glib.swift
+//  LyricsX - https://github.com/ddddxxx/LyricsX
+//
+//  This Source Code Form is subject to the terms of the Mozilla Public
+//  License, v. 2.0. If a copy of the MPL was not distributed with this
+//  file, You can obtain one at https://mozilla.org/MPL/2.0/.
+//
+
+#if os(Linux)
+
+import Foundation
+import playerctl
+
+func gproperty<T, R>(_ ptr: UnsafeMutablePointer<T>, name: String, transform: (UnsafeMutablePointer<GValue>) -> R) -> R {
+    ptr.withMemoryRebound(to: GObject.self, capacity: 1) {
+        var value = GValue()
+        g_object_get_property($0, name, &value)
+        return transform(&value)
+    }
+}
+
+
+class GEventLoop {
+    
+    static private(set) var loop: OpaquePointer? /* GMainLoop* */ = nil
+    static private(set) var running = false
+    
+    static func start() {
+        if !running {
```

Comment:
Yes, it is not thread safe.
Locking will increase the overhead for each initialization, and using `DispatchQueue` will create a new thread.
It is indeed not a good idea to let users ensure that they are in the same thread (or queue) every time they create a player.
Also, users may already have their own `GMainLoop`, in which case this `GMainLoop` is unnecessary.
Perhaps it is better to let users run `GMainLoop`, but it will increase the cost of library usage.

Is there a better way? Or I change to use `DispatchQueue` to execute `start` and `quit` as a compromise.

Reply:
The best solution may be [integrating GMainLoop into CFRunloop](https://github.com/GNOME/gtk/blob/master/gdk/macos/gdkmacoseventsource.c), but that's far too complex (maybe add a // TODO?). A serial queue sounds good enough to me.